src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -334,10 +409,7 @@ private void AddBindings(ArrayBuilder<BoundStatement> sectionBuilder, ImmutableA
                     {
                         var source = kv.Key;
                         var dest = kv.Value;
-                        var rewriter = this.LocalRewriter;
-                        sectionBuilder.Add(_factory.ExpressionStatement(
-                            rewriter.MakeStaticAssignmentOperator(
-                                _factory.Syntax, rewriter.VisitExpression(dest), rewriter.VisitExpression(source), RefKind.None, dest.Type, false)));
+                        sectionBuilder.Add(_factory.Assignment(dest, source));
```

Comment:
The nodes were created by the caller. Specifically, this method is called by the private method `LowerDecisionTree`, which is called from `LowerPatternSwitch`:

```cs
            private void LowerPatternSwitch(BoundExpression loweredExpression, BoundPatternSwitchStatement node, ArrayBuilder<BoundStatement> loweredDecisionTree)
            {
                var decisionTree = LowerToDecisionTree(loweredExpression, node);
                LowerDecisionTree(loweredExpression, decisionTree, loweredDecisionTree);
            }
```

So the nodes are created during lowering, and then assembled into the assignment here as part of lowering.